### PR TITLE
[Fix] Inject LAContext to the evaluateAuthentication(...)

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?

There were breaking changes in the DM:  evaluateAuthentication(...) method got a new parameter.

Needs releases with:

- [ wireapp/wire-ios-data-model/pull/1105 ]

### CODE REVIEW

File: Wire-iOS Share Extension/ShareExtensionViewController.swift [hunk 1]
```diff
@@ -496,13 +497,16 @@ final class ShareExtensionViewController: SLComposeServiceViewController {
             scenario = .screenLock(requireBiometrics: sharingSession.appLockController.config.useBiometricsOrCustomPasscode)
         }
         
-        sharingSession.appLockController.evaluateAuthentication(scenario: scenario,
-                                                                description: "share_extension.privacy_security.lock_app.description".localized)
+        guard let appLockController = sharingSession.appLockController as? AppLockController else {
+            return
+        }
```
Comment (johnxnguyen) [inline Wire-iOS Share Extension/ShareExtensionViewController.swift:502]: Why do you need to cast this?

Comment (KaterinaWire) [reply Wire-iOS Share Extension/ShareExtensionViewController.swift:502]: Otherwise, I need to use the context as a parameter.

Comment (KaterinaWire) [reply Wire-iOS Share Extension/ShareExtensionViewController.swift:502]: The default value exists only for `AppLockController`

Comment (johnxnguyen) [reply Wire-iOS Share Extension/ShareExtensionViewController.swift:502]: You could add an extension that does this for you:

```swift
extension AppLockType {

  evaluateAuthentication(scenario: AuthenticationScenario, description: String) {
    evaluateAuthentication(scenario: AuthenticationScenario, context: LAContext(), description: String) {
  }

}
```

Comment (KaterinaWire) [reply Wire-iOS Share Extension/ShareExtensionViewController.swift]: I added this extension to the data model PR

**Review by johnxnguyen — APPROVED** [review]

File: Wire-iOS Share Extension/ShareExtensionViewController.swift [hunk 2]
```diff
@@ -502,7 +503,7 @@ final class ShareExtensionViewController: SLComposeServiceViewController {
             DispatchQueue.main.async {
                 if case .granted = result {
                     self?.localAuthenticationStatus = .granted
-                    try? self?.sharingSession?.unlockDatabase(with: context)
+                    try? self?.sharingSession?.unlockDatabase(with: context as! LAContext)
```
Comment (David-Henner) [inline Wire-iOS Share Extension/ShareExtensionViewController.swift:506]: should you try to safely unwrap `context` ?

Comment (KaterinaWire) [reply Wire-iOS Share Extension/ShareExtensionViewController.swift:506]: In this case I'm sure that context will be `LAContext`. What is your suggestion?

Comment (KaterinaWire) [reply Wire-iOS Share Extension/ShareExtensionViewController.swift:506]:
```
if let context = context as? LAContext {
                        try? self?.sharingSession?.unlockDatabase(with: context)
  }
```
?

Comment (David-Henner) [reply Wire-iOS Share Extension/ShareExtensionViewController.swift:506]: I was thinking something like that yep. But if you think force unwrapping is safe in this context, go ahead :)

**Review by David-Henner — APPROVED** [review]

**Comment by David-Henner** [comment]
Are you planning on doing a release train before merging ?

**Comment by KaterinaWire** [comment]
> Are you planning on doing a release train before merging ?

Yes

File: Wire-iOS Share Extension/ShareExtensionViewController.swift (hunk 2 above)
Comment (KaterinaWire) [reply Wire-iOS Share Extension/ShareExtensionViewController.swift]: I added changes

**Review by typfel — APPROVED** [review]

**Pull request merged**